### PR TITLE
feat: contacts UI update

### PR DESCRIPTION
**Pull request by bryce-mcmath**
# Summary of Changes

Various UI updates for the contact list, contact details, and what are contacts screens. Also capitalized all instances of Contact / Contacts as it was inconsistently capitalized previously.

### Here is the old contact list:
contacts_list_old

### Here is the new contact list:
contacts_list_new

### Here is the new contact details:
contact_details

### Here is the new what are contacts screen:
what_are_contacts

### Here is the new empty contacts list:
empty_contacts

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/core/App/localization/en/index.ts [hunk 1]
```diff
@@ -103,8 +103,8 @@ const translation = {
     "Message1035": "Credential offer could not be found.",
     "Title1036": "Unable to fetch wallet credentials",
     "Message1036": "Wallet credentials could not be found",
-    "Title1037": "Unable to remove contact.",
-    "Message1037": "There was a problem while removing the contact.",
+    "Title1037": "Unable to remove Contact.",
```
Comment (jleach) [inline packages/legacy/core/App/localization/en/index.ts:106]: Why capital "C" on contact?

File: packages/legacy/core/App/components/listItems/ContactListItem.tsx [hunk 2]
```diff
@@ -41,21 +64,62 @@ const ContactListItem: React.FC<Props> = ({ contact, navigation }) => {
...
+          text: userLabel + actionLabel + '.',
+          createdAt: record.updatedAt || record.createdAt,
+        }
+      })
+    )
+
+    transformedMessages.push(
+      ...proofs.map((record: ProofExchangeRecord) => {
+        const role = getProofEventRole(record)
+        const userLabel = role === Role.me ? t('Chat.UserYou') + ' ' : ''
```
Comment (jleach) [inline packages/legacy/core/App/components/listItems/ContactListItem.tsx]: I think `+` is old skool. Better to `${xxx} `.

File: packages/legacy/core/App/components/listItems/ContactListItem.tsx [hunk 3]
```diff
@@ -41,21 +64,62 @@ const ContactListItem: React.FC<Props> = ({ contact, navigation }) => {
...
+        const actionLabel = t(getProofEventLabel(record) as any)
+
+        return {
+          text: userLabel + actionLabel + '.',
```
Comment (jleach) [inline packages/legacy/core/App/components/listItems/ContactListItem.tsx]: Same `+`.

File: packages/legacy/core/App/components/listItems/ContactListItem.tsx [hunk 4]
```diff
@@ -41,21 +64,62 @@ const ContactListItem: React.FC<Props> = ({ contact, navigation }) => {
...
+      }
+    })
+
+    transformedMessages.push(
+      ...credentials.map((record: CredentialExchangeRecord) => {
+        const role = getCredentialEventRole(record)
+        const userLabel = role === Role.me ? t('Chat.UserYou') + ' ' : ''
+        const actionLabel = t(getCredentialEventLabel(record) as any)
+        return {
+          text: userLabel + actionLabel + '.',
```
Comment (jleach) [inline packages/legacy/core/App/components/listItems/ContactListItem.tsx]: `+`

File: packages/legacy/core/App/components/listItems/ContactListItem.tsx [hunk 5]
```diff
@@ -41,21 +64,62 @@ const ContactListItem: React.FC<Props> = ({ contact, navigation }) => {
...
+      return {
+        text: record.content,
+        createdAt: record.updatedAt || record.createdAt,
+      }
+    })
+
+    transformedMessages.push(
+      ...credentials.map((record: CredentialExchangeRecord) => {
+        const role = getCredentialEventRole(record)
+        const userLabel = role === Role.me ? t('Chat.UserYou') + ' ' : ''
```
Comment (jleach) [inline packages/legacy/core/App/components/listItems/ContactListItem.tsx]: `+`

File: packages/legacy/core/App/components/listItems/ContactListItem.tsx [hunk 6]
```diff
@@ -1,24 +1,47 @@
...
 interface Props {
   contact: ConnectionRecord
   navigation: StackNavigationProp<ContactStackParams, Screens.Contacts>
 }
 
 const ContactListItem: React.FC<Props> = ({ contact, navigation }) => {
   const { t } = useTranslation()
-  const { TextTheme, ColorPallet } = useTheme()
+  const { TextTheme, ColorPallet, ListItems } = useTheme()
+  const connectionId = contact.id
```
Comment (jleach) [inline packages/legacy/core/App/components/listItems/ContactListItem.tsx]: I don't think `connectionId` is used anywhere besides the 3 following lines. Why not just `useBasicMessagesByConnectionId(contact.id)`

File: packages/legacy/core/App/components/listItems/ContactListItem.tsx [hunk 7]
```diff
@@ -41,21 +64,62 @@ const ContactListItem: React.FC<Props> = ({ contact, navigation }) => {
...
+        const userLabel = role === Role.me ? t('Chat.UserYou') + ' ' : ''
+        const actionLabel = t(getProofEventLabel(record) as any)
+
+        return {
+          text: userLabel + actionLabel + '.',
+          createdAt: record.updatedAt || record.createdAt,
+        }
+      })
+    )
+    setMessage(transformedMessages.sort((a: any, b: any) => b.createdAt - a.createdAt)[0]?.text || '')
```
Comment (jleach) [inline packages/legacy/core/App/components/listItems/ContactListItem.tsx]: `??` or `||`?

**Review by jleach — CHANGES_REQUESTED** [review]

File: packages/legacy/core/App/localization/en/index.ts (hunk 1 above)
Comment (bryce-mcmath) [reply packages/legacy/core/App/localization/en/index.ts:106]: Instructions from UI/UX team

File: packages/legacy/core/App/components/listItems/ContactListItem.tsx (hunk 7 above)
Comment (bryce-mcmath) [reply packages/legacy/core/App/components/listItems/ContactListItem.tsx]: ??

File: packages/legacy/core/App/components/listItems/ContactListItem.tsx (hunk 6 above)
Comment (bryce-mcmath) [reply packages/legacy/core/App/components/listItems/ContactListItem.tsx]: done

File: packages/legacy/core/App/components/listItems/ContactListItem.tsx (hunk 2 above)
Comment (bryce-mcmath) [reply packages/legacy/core/App/components/listItems/ContactListItem.tsx]: done

**Review by jleach — APPROVED** [review]

**Pull request merged**